lib/iris/fileformats/netcdf/loader.py

```diff
@@ -624,6 +624,34 @@ def inner(cf_datavar):
                             break
                 return match
 
+            result = inner
```

Comment:
I really think we should **combine these 2 branches**, 
i.e. the `len(constraints) == 1` one and the `len(constraints) > 1` one.
Since the code is so similar it totally breaks DRY!
(I suspect you intended to do this + forgot??)

In fact, the new 'branch' code is already covering the ==1 case, since a single constraint is presented as a list of 1 item.  I've **confirmed that this works**. 
However, in the process I also found that the `len(constraints) > 0` test is essential, because otherwise we return a function which rejects all the vars (!)

Actually, for that reason alone, I think it's worth testing the no-constraints case, which happens not to be tested at present in `test__translate_constraints_to_var_callback.py` : see separate comment